tools/src/main/scala/com/nvidia/spark/rapids/tool/profiling/AutoTuner.scala

```diff
@@ -0,0 +1,435 @@
+    } else {
+      // Recommendation for 'spark.task.resource.gpu.amount' based on num of cpu cores and
+      // 'spark.rapids.sql.concurrentGpuTasks' based on gpu memory
+      val numGpus: Int = systemProps.gpu_props.count
+      val numCores: Int = bestConfig.get_executor_cores
+
+      val taskResourceGpu: Double = 1.0 / numCores
+      val gpuMemoryNum: Long = convertFromHumanReadableSize(systemProps.gpu_props.memory)
+      val concurrentGpuTasks: Int = Math.min(
+        gpuMemoryNum * DEFAULT_CONCURRENT_GPU_TASKS_MULTIPLIER,
+        MAX_CONCURRENT_GPU_TASKS).toInt
+
+      bestConfig.set_task_resource_gpu(taskResourceGpu)
+      bestConfig.set_concurrent_gpu_tasks(concurrentGpuTasks)
+
+      if(numCores < concurrentGpuTasks) {
+        comments :+= s"For the given GPU, number of CPU cores is very low. It should be" +
+          s" at least equal to concurrent gpu tasks i.e. $concurrentGpuTasks."
+      }
+
+      // Recommendation for 'spark.executor.memoryOverhead', 'spark.executor.memoryOverheadFactor'
+      // or 'spark.kubernetes.memoryOverheadFactor' based on cluster scheduler, spark
+      // version. See recommendMemoryOverhead() for the calculation used.
+      getSparkProperty(app, "spark.rapids.memory.pinnedPool.size") match {
+        case Some(pinnedPoolSize) =>
+          val sparkMaster = getSparkProperty(app, "spark.master")
+          if (sparkMaster.contains("k8s")) {
+            if (compareSparkVersion(app.appInfo.head.sparkVersion, "3.3.0") > 0) {
+              if (getSparkProperty(app, "spark.executor.memoryOverheadFactor").isEmpty) {
+                comments :+= "'spark.executor.memoryOverheadFactor' must be set " +
+                  "if using 'spark.rapids.memory.pinnedPool.size'"
+              }
+            } else {
+              if (getSparkProperty(app, "spark.kubernetes.memoryOverheadFactor").isEmpty) {
+                comments :+= "'spark.kubernetes.memoryOverheadFactor' must be set " +
+                  "if using 'spark.rapids.memory.pinnedPool.size'"
+              }
+            }
+          } else if (sparkMaster.contains("yarn")) {
+            if (getSparkProperty(app, "spark.executor.memoryOverhead").isEmpty) {
+              val memoryOverhead = recommendMemoryOverhead(pinnedPoolSize,
+                bestConfig.get_executor_memory)
+              bestConfig.set_executor_memory_overhead(convertToHumanReadableSize(memoryOverhead))
+            }
+          }
+
+        case None =>
+          bestConfig.set_pinned_pool_size(DEFAULT_PINNED_POOL_SIZE)
+          val memoryOverhead = recommendMemoryOverhead(DEFAULT_PINNED_POOL_SIZE,
+            bestConfig.get_executor_memory)
+          bestConfig.set_executor_memory_overhead(convertToHumanReadableSize(memoryOverhead))
+      }
+    }
+  }
+
+  def getRecommendedProperties(): (Seq[RecommendedPropertyResult],
+    Seq[RecommendedCommentResult])= {
+    val systemProps = parseSystemInfo(workerInfo)
```

Comment:
it looks like is this assuming the worker info is always there, what is they don't run the script or pass the parameter?  we should check for existence of file and if not there at least error or use another method

Reply:
1. `parseSystemInfo()` returns `null` if the worker-info is not present or contents are empty.
2. Now, handling of `null` value for `systemProps` is made by the individual recommender functions independently. i.e `recommendSparkProperties` and `recommendGpuProperties`